packages/node_modules/@webex/plugin-meetings/src/meetings/index.js

```diff
@@ -204,35 +216,33 @@ export default class Meetings extends SparkPlugin {
         parent: this.spark
       }
     );
-    MeetingCollection.set(meeting);
 
-    // Moved getMeetingInfo later as it takes more time for the network request
-    // meanwhile if the client is flooded with mercury event it wont cause any issues
-    this.meetingInfo
+    return this.meetingInfo
       .fetchMeetingInfo(MeetingsUtil.extractDestination(destination, type), type)
       .then((info) => {
         meeting.parseMeetingInfo(info);
         meeting.meetingInfo = info;
       })
       .catch((err) => {
-        if (!meeting.sipUri) {
-          meeting.setSipUri(destination);
-        }
         console.error(err);
       })
-      .finally(() => {
+      .then(() => {
         if (!meeting.sipUri) {
           meeting.setSipUri(destination);
         }
+        // TODO: check if we have to move this to parser
+        const meetingAddedType = MeetingsUtil.getMeetingAddedType(type);
+
+        // We typically shouldn't need to trigger both and event and return a promise.
+        // Is this a special case? We want to make the public API usage as simple as possible.
+        this.trigger(EVENT_TRIGGERS.MEETING_ADDED, {
+          meeting,
+          type: meetingAddedType
+        });
+
+        return MeetingCollection.set(meeting);
```

Comment:
Can we move this on the top before fetching the meeting INfo

Reply:
This is required to avoid creating multiple meeting object if there is a flood of delta or mercury event